File: Difficulty: Easy/Second Largest/second-largest.cpp
class Solution {
  public:
    int getSecondLargest(vector<int> &arr) {
        // code here
        int n=arr.size();
        int largest=INT_MIN;
        int s_largest=INT_MIN;
        for(int x:arr){
            if(x>largest){
                s_largest=largest;
                largest=x;
            }
            else if(largest>x && x>s_largest){
                s_largest=x;
            }
        }
        if (s_largest==INT_MIN ) return -1;
        else return s_largest;
        
    }
};

File: Difficulty: Easy/Second Largest/second-largest.java
class Solution {
    public int getSecondLargest(int[] arr) {
        // code here
        int largest=Integer.MIN_VALUE;
        int S_largest=Integer.MIN_VALUE;
        for(int i=0;i<arr.length;i++){
            if(arr[i]>largest){
                S_largest=largest;
                largest=arr[i];
            }
            else if(largest > arr[i] && arr[i] > S_largest){
                S_largest=arr[i];
            }
        }
        if(S_largest==Integer.MIN_VALUE) return -1;
        else return S_largest;
    }
}